rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -368,21 +372,41 @@ uint64_t SqliteStorage::get_bagfile_size() const
 
 void SqliteStorage::initialize()
 {
-  std::string create_stmt = "CREATE TABLE topics(" \
+  std::string create_stmt = "CREATE TABLE schema(" \
+    "schema_version INTEGER PRIMARY KEY," \
+    "ros_distro TEXT NOT NULL);";
```

Comment:
Why not store these two keys in the metadata table?

Reply:
There are multiple reasons to not store these two keys in the metadata table.
1. To not make a mess and not store everything in one table.
2.  Schema version should be first what should be read from table and upon of it value we could make a decision what and in how to read. For instance even in current implementation we have a logic that if `schema_version < 3` we will not even try to read metadata since on those versions metadata table doesn't exists.
3. According to the `update_metadata` API https://github.com/ros2/rosbag2/pull/1149. We will call `update_metadata` twice one time right after opening storage and second time just before closing bag. Since currently we don't have update API for fields in DB perhaps we will store 2 records for metadata and store another fields repeatedly looks odd.
4. To be able to propagate changes related for storing `ROS_DISTRO` name to the Humble. If we would store it inside metadata table we would likely need to make changes which is breaking ABI and API from #1149. Note that we don't have valid instance of metadata object during bag initialization and `storage::open()` call.

Reply:
`ROS_DISTRO` does seem like an appropriate addition to `BagMetadata`, to me. The `schema_version` on the other hand is unique to the storage implementation and so shouldn't be part of generic metadata, so it makes sense for it to live somewhere else.